### PR TITLE
feat(python): support auto-retry of failed network requests

### PR DESCRIPTION
**Pull request by evansims**
## Description

This PR rolls up the changes in the following PRs from the Python SDK and updates the generator to incorporate those updates:

- https://github.com/openfga/python-sdk/pull/78
- https://github.com/openfga/python-sdk/pull/79

To summarize, these changes introduce auto-retry support for failed requests (429, 500, 502, ...) to the `/oauth/token` auth endpoint, and enhance existing auto-retry of failed requests to the OpenFGA server to cover 5xx series error responses. In both cases, 501 is excluded from retries.

This PR also updates the makefile to generate a more robust code coverage report when running unit tests of the Python SDK. There is 100% coverage of these changes.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: config/clients/python/template/oauth2.mustache [hunk 1]
```diff
@@ -32,25 +56,59 @@ class OAuth2Client:
...
+            )
+            else 0
+        )
+
+        for attempt in range(max_retry + 1):
+            raw_response = await client.POST(
+                token_url, headers=headers, post_params=post_params
+            )
+
+            if 500 <= raw_response.status <= 599 or raw_response.status == 429:
```
Comment (rhamzeh) [inline config/clients/python/template/oauth2.mustache:94]: It seems like we're raising Auth errors on 404?

Can you check:
https://github.com/openfga/js-sdk/blob/main/common.ts#L155-L175
It doesn't handle 501s, but it covers the rest of the cases.

For the special cases of oauth token exchange, we then catch the error, whatever it is and mark it as "FgaApiAuthenticationError"
https://github.com/openfga/js-sdk/blob/main/credentials/credentials.ts#L163-L169

File: config/clients/python/template/oauth2_test_sync.mustache [hunk 2]
```diff
@@ -97,3 +97,145 @@ class TestOAuth2Client(IsolatedAsyncioTestCase):
...
+}
+        """
+        mock_request.return_value = mock_response(response_body, 403)
+
+        credentials = Credentials(
+            method="client_credentials",
+            configuration=CredentialConfiguration(
+                client_id="myclientid",
+                client_secret="mysecret",
+                api_issuer="www.testme.com",
```
Comment (rhamzeh) [inline config/clients/python/template/oauth2_test_sync.mustache:121]:
```suggestion
                api_issuer="https://tokenissuer.{{sampleApiDomain}}",
```

Comment (rhamzeh) [reply config/clients/python/template/oauth2_test_sync.mustache:121]: or `https://issuer.{{sampleApiDomain}}`, both work. Just to remain consistent with our other SDKs, in general all outgoing requests are part of "{{sampleApiDomain}}", which is currently set to `fga.example`, which even if we set mocking incorrectly will never resolve to an actual domain

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: config/clients/python/template/oauth2_test_sync.mustache (hunk 2 above)
Comment (evansims) [reply config/clients/python/template/oauth2_test_sync.mustache:121]: Sure thing, I was just going off of what the existing Python tests were using — you good for me to just update them all in this PR, or would you prefer I cut a separate one?

Comment (rhamzeh) [reply config/clients/python/template/oauth2_test_sync.mustache:121]: I'm fine with either, but if this PR is easier, let's do that

Comment (evansims) [reply config/clients/python/template/oauth2_test_sync.mustache]: Sounds good; PR updated 😄

**Pull request merged**